src/QueryEditor.tsx

```diff
@@ -0,0 +1,706 @@
+
+  //   data.map((incomingT: any) => {
+  //     newTagsValues.push(incomingT.value);
+  //   });
+
+  //   setTagValue(data);
+  //   onApplyQueryChange({ ...query, tagsValues: newTagsValues });
+  // };
+
+  // set tenancyName in case dashboard was created with version 4.x
+  if (query.tenancy && !hasLegacyTenancy && !query.tenancyName) {
+      query.tenancyName = query.tenancy;  
+      setTenancyValue(query.tenancy);
+      setHasLegacyTenancy(true);
+  }
+
+  // set compartmentName in case dashboard was created with version 4.x
+  if (!query.compartmentName && query.compartment && !hasLegacyCompartment) {
+    if (!query.tenancy && tmode === TenancyChoices.multitenancy) {
+      console.log("query.tenancy is empty");
+      return null;
+    }
+    datasource.getCompartments(query.tenancy).then(response => {
+      if (response) {
+        let found = false;
+        response.forEach((item: any) => {
+          if (!found && item.ocid === query.compartment) {
+            found = true; 
+            query.compartmentName = item.name;
+          } else if (!found) {
+            query.compartmentName = query.compartment;
+          }           
+        });
+      } else {
+          query.compartmentName = query.compartment;    
+      }
+      setCompartmentValue(query.compartmentName);
+      setHasLegacyCompartment(true);
+    });
+}
+
+
+  return (
+    <>
+      <FieldSet>
+        <InlineFieldRow>
+          {tmode === TenancyChoices.multitenancy && (
+            <>
+              <InlineField label="TENANCY" labelWidth={20}>
+                <SegmentAsync
+                  className="width-42"
+                  allowCustomValue={false}
+                  required={true}
+                  loadOptions={getTenancyOptions}
+                  value={tenancyValue}
+                  placeholder={QueryPlaceholder.Tenancy}
+                  onChange={(data) => {
+                    onTenancyChange(data);
+                  }}
+                />
+              </InlineField>
+            </>
+          )}
+          {tmode === TenancyChoices.single && (
+            <>
+        <InlineField label="TENANCY" labelWidth={20}>
+          <CustomInput className="width-14" value={"DEFAULT/"} readOnly />
+        </InlineField>
+            </>
+          )}          
+        </InlineFieldRow>
+        <InlineFieldRow>
+          <InlineField label="REGION" labelWidth={20}>
+            <SegmentAsync
+              className="width-14"
+              allowCustomValue={true}
+              required={true}
+              loadOptions={getSubscribedRegionOptions}
+              value={regionValue}
+              placeholder={QueryPlaceholder.Region}
+              onChange={(data) => {
+                onRegionChange(data);
+              }}
+            />
+          </InlineField>
+          <InlineField label="COMPARTMENT" labelWidth={20}>
+            <SegmentAsync
+              className="width-28"
```

Comment:
Just a small suggestion, we can get rid of this and then increase column size slightly to use the space. Ignore if you think it's not necessary

![image](https://github.com/oracle/oci-grafana-metrics/assets/125669121/769de4f5-65ec-48ed-96ee-bca0ec936c53)

Reply:
looks like it only accepts multiple of 14, and 42 will be too big :(